debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java

```diff
@@ -145,6 +145,30 @@ public Config validate(Map<String, String> connectorConfigs) {
                         final String errorMessage = "Postgres roles LOGIN and REPLICATION are not assigned to user: " + connection.username();
                         LOGGER.error(errorMessage);
                     }
+                    // check replication slot
+                    final String slotName = config.slotName();
+                    if (connection.prepareQueryAndMap(
+                            "SELECT slots.active AS slot_active" +
+                                    " FROM pg_replication_slots AS slots" +
+                                    " INNER JOIN pg_stat_activity AS connections" +
+                                    " ON slots.datoid=connections.datid AND slots.active_pid=connections.pid" +
+                                    " WHERE slots.slot_name = ?" +
+                                    " AND connections.application_name != ?",
+                            statement -> {
+                                statement.setString(1, slotName);
+                                statement.setString(2, "DBZ#" + config.getConfig().getString("name"));
```

Comment:
So thinking about the case where multiple connectors are using the same connector and slot names, but run on different KC clusters. In that case, we'd pass validation here, but still the 2nd connector to start would fail because it couldn't claim the slot. So this should be fine?

Reply:
That's a very good point. It should not validate, I think. And we should call that same validation on Connector#start maybe instead the active replication slot check?
I need to wrap my head around that scenario with multiple clusters.
Now that I have a setup to actually test it with testcontainers, let me add a test for that case.